spec/defines/admin_spec.rb

```diff
@@ -50,6 +50,29 @@
               .with_unless('/usr/bin/false')
           end
         end
+
+        context 'default parameters' do
```

Comment:
`spec/defines/admin_spec.rb` is used to test that the `pulpcore::admin` class compiles properly under various params.

In all the examples here, we're testing `pulpcore::admin { 'help' }` as you can see when we defined the `title` for the class here: https://github.com/theforeman/puppet-pulpcore/blob/master/spec/defines/admin_spec.rb#L7 , basically using `help` as a dummy admin command we might to run, and then testing that it behaves properly with various combinations of `unless`, `refreshonly`, etc.

So probably this isn't the right test to use for our purposes. I'll suggest alternatives in another comment.

Reply:
In `spec/acceptance/basic_spec.rb` I think we could test that the admin user got created after running puppet-apply, using the same logic to test as in your `unless` condition, adding after L55:

```
describe command("DJANGO_SETTINGS_MODULE=pulpcore.app.settings python3-django-admin dumpdata | grep "auth.user"") do
  its(:exit_status) { is_expected.to eq 0 }
```

We have to fix the issue `NameError: name 'CONTENT_ORIGIN' is not defined` that we saw previously before this will work.

Reply:
As @wbclark said, this probably belongs in `spec/classes/pulpcore_spec.rb`